### PR TITLE
WC: dapp warnings

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes APP-879 APP-878 APP-877

## Screen recordings / screenshots
scam: https://cloud.skylarbarrera.com/Screen-Shot-2023-10-23-14-02-29.97.png
verified: https://cloud.skylarbarrera.com/Screen-Shot-2023-10-23-14-02-48.12.png

### CODE REVIEW

**Comment by skylarbarrera** [comment]
auth requests 
scam: https://cloud.skylarbarrera.com/Screen-Shot-2023-10-24-21-52-18.53.png

no scam: https://cloud.skylarbarrera.com/Screen-Shot-2023-10-24-21-53-16.14.png

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good on both OS's 👍🏽

File: src/walletConnect/index.tsx [hunk 1]
```diff
@@ -1002,13 +1006,22 @@ export async function onAuthRequest(event: Web3WalletTypes.AuthRequest) {
     }
   };
 
+  // need to prefetch dapp metadata since portal is static
+  const url =
+    // @ts-ignore Web3WalletTypes.AuthRequest type is missing VerifyContext
+    event?.verifyContext?.verifyUrl || event.params.requester.metadata.url;
```
Comment (dereknelson) [inline src/walletConnect/index.tsx:1012]: nit but i prefer `??` over `||` when optional

pretty sure they're functionally equivalent

File: src/screens/WalletConnectApprovalSheet.js [hunk 2]
```diff
@@ -253,6 +256,18 @@ export default function WalletConnectApprovalSheet() {
 
   const { dappName, dappUrl, dappScheme, imageUrl, peerId } = meta;
 
+  const verifiedData = params?.verifiedData;
+  const { data: metadata } = useDappMetadata({
+    url: verifiedData?.verifyUrl || dappUrl,
```
Comment (dereknelson) [inline src/screens/WalletConnectApprovalSheet.js:261]: nit x2

**Review by dereknelson — APPROVED** [review]
i indeed was warned. let's get this (user) money (proteccted)

File: src/walletConnect/index.tsx (hunk 1 above)
Comment (skylarbarrera) [reply src/walletConnect/index.tsx:1012]: `??` is a lil more restrictive, here verifyUrl can be an empty string`""`

`||` will fallback
`??` will not fallback and use the empty string

**Review by benisgold — APPROVED** [review]

**Pull request merged**